### PR TITLE
Enable retries for SSL_ERROR_WANT_READ

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

This PR enables retries for `SSL_read` when the error returned is `SSL_ERROR_WANT_READ`.

This appears to help when a child attempts to connect and stream to a parent over a bad network line when using SSL.

##### Test Plan

Would like some feedback on testing this scenario.

1) Create a child and parent from master on different machines.
2) Establish between them an SSL streaming configuration.
3) Use [this utility](https://gist.github.com/denilsonsa/5176e1c9b6a119594ce0) to simulate a bad network connection on the parent. For my tests I used `sudo ./slow -d enp4s0f4u2u4 3G -l 600ms -p 10%` (replace enp4s0f4u2u4 with your network adapter).
4) Start child/parent and observe the logs. On my tests the child was throwing `SSL_read() returned -1 bytes, SSL error 2` (Error 2 is `SSL_ERROR_WANT_READ`). The parent side had similar errors, going into a connect/disconnect loop.
5) Apply the patch on both parent and child. Re-try and notice if now there is a stable streaming connection between them.

##### Additional Information

I would like some more tests on this in case my setup is an isolated instance.

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Comment by ilyam8** [comment]
@MrZammler according to @ktsaou [retrying reading or writing corrupts the data](https://github.com/netdata/netdata/pull/14026).

**Comment by MrZammler** [comment]
Yes, I saw that, thanks. I didn't enable `SSL_ERROR_WANT_WRITE` though, and indeed, we need to test it, but I think at least the retry for `SSL_ERROR_WANT_READ` is needed...

File: libnetdata/socket/socket.c [hunk 1]
```diff
@@ -925,11 +925,11 @@ ssize_t netdata_ssl_read(SSL *ssl, void *buf, size_t num) {
 
     int bytes, err, retries = 0;
 
-    //do {
+    do {
         bytes = SSL_read(ssl, buf, (int)num);
         err = SSL_get_error(ssl, bytes);
         retries++;
-    //} while (bytes <= 0 && (err == SSL_ERROR_WANT_READ));
+    } while (err == SSL_ERROR_WANT_READ);
```
Comment (ktsaou) [inline libnetdata/socket/socket.c:932]: The way this is now, if the first time `bytes` is positive, the next iteration will overwrite data.

So, if there is possibility to have `SSL_ERROR_WANT_READ` while we have read some data, these data will now be lost (overwritten at like 929 on the second iteration).

I don't know if it is possible to get `SSL_ERROR_WANT_READ` with positive bytes though.

**Review by ktsaou — CHANGES_REQUESTED** [review]

File: libnetdata/socket/socket.c (hunk 1 above)
Comment (underhood) [reply libnetdata/socket/socket.c:932]: wouldn't SSL_MODE_AUTO_RETRY suffice?
in other case should you not wait for bytes on the socket first?

Comment (ktsaou) [reply libnetdata/socket/socket.c:932]: `SSL_MODE_AUTO_RETRY` probably yes. I am not familiar with SSL though. Can you make it happen? We need the same on the writing side I think...

Comment (underhood) [reply libnetdata/socket/socket.c:932]: Yeah, IMHO the corrupted data @ilyam8 mentioned above are just caused by wrong usage of OpenSSL

Comment (MrZammler) [reply libnetdata/socket/socket.c:932]: > wouldn't SSL_MODE_AUTO_RETRY suffice? in other case should you not wait for bytes on the socket first?

Timo, we set this on `netdata_ssl_client_ctx`, should we also on `netdata_ssl_srv_ctx` ?

Comment (underhood) [reply libnetdata/socket/socket.c:932]: I have not checked the code on `srv_ctx` but if you want same behavior (openssl dont bothering you with handling retries ) I suppose yes.

**Comment by thiagoftsm** [comment]
A quick update, after an initial chat with @MrZammler , because I was receiving these errors on my environment while I was working with my PR:

```sh
2022-12-19 12:11:13: netdata ERROR : STREAM_SENDER[archlinux] : (0935@libnetdata/socket/so:netdata_ssl_rea): SSL_read() returned -1 bytes, SSL error 2 (errno 11, Resource temporarily unavailable)
2022-12-19 12:11:13: netdata INFO  : STREAM_SENDER[archlinux] : (0502@streaming/rrdpush.c :connect_to_one_): STREAM archlinux: connecting to '192.168.122.1:19999' (default port: 19999)...
2022-12-19 12:11:13: netdata ERROR : STREAM_SENDER[archlinux] : (0427@streaming/sender.c  :rrdpush_sender_): STREAM archlinux [send to 192.168.122.1:19999]: remote server rejected this stream, the host we are trying to stream is already streamed to it - will retry in 120 secs, at 2022-12-19 12:13:13
2022-12-19 12:11:14: netdata ERROR : GLOBAL_STATS : (0420@streaming/rrdpush.c :rrdset_done_pus): STREAM archlinux [send]: not ready - collected metrics are not sent to parent.
```

I tested this PR on the same environment and I did not have more the issue. I tested this PR only on children side for while.

**Comment by Dim-P** [comment]
@MrZammler I was just testing to see if this PR fixes https://github.com/netdata/netdata/issues/14129 and it seems it reduces the frequency that it happens when I refresh the page, but it still happens sometimes:

Capture

But that might be something else, not 100% sure if https://github.com/netdata/netdata/issues/14129 is exactly the same issue as this one. Error is `SSL_ERROR_WANT_WRITE` in my case and I am also getting these info messages:
`2022-12-13 22:20:44: netdata INFO  : WEB_SERVER[static1] : 0 SSL Handshake error (error:00000006:lib(0)
:func(0):EVP lib) on socket 67`

Also, a small suggestion; the following error message is a bit confusing now, since we modify the amount of bytes returned after the error message: 
```
error("SSL_write() returned %d bytes, SSL error %d", bytes, err);
 if (err == SSL_ERROR_WANT_WRITE || err == SSL_ERROR_WANT_READ) {
            bytes = 0;
 }
```

I would suggest maybe adding another `error` message inside `if(...){...}` so that we know when bytes are set to 0, or change the existing one to something like:
`error("SSL_write() returned %d bytes%s, SSL error %d", bytes, err == SSL_ERROR_WANT_WRITE || err == SSL_ERROR_WANT_READ ? "(will be set to 0)" : "", err);`

**Comment by Habetdin** [comment]
Leaving some debug info I've posted [here](https://github.com/netdata/netdata/issues/14129#issuecomment-1360379206) just to make this topic searchable:

I have the `SSL error 2` too:
```
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: initializing communication...
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: waiting response from remote netdata...
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: established link with negotiated capabilities: VCAPS HLABELS CLAIM CLABELS COMPRESSION FUNCTIONS REPLICATION BINARY 
2022-12-21 00:39:13: netdata ERROR : STREAM_SENDER[CHILD_NODE_HOSTNAME] : Clearing stream_collected_metrics flag in charts of host CHILD_NODE_HOSTNAME
2022-12-21 00:39:13: netdata INFO  : STREAM_SENDER[CHILD_NODE_HOSTNAME] : STREAM CHILD_NODE_HOSTNAME [send to <PARENT_NODE_IP>:20000]: enabling metrics streaming...
2022-12-21 00:39:13: netdata INFO  : PLUGINSD[nfacct] : STREAM CHILD_NODE_HOSTNAME [send]: sending metrics to parent...
2022-12-21 00:39:14: netdata LOG FLOOD PROTECTION too many logs (201 logs in 10 seconds, threshold is set to 200 logs in 1200 seconds). Preventing more logs from process 'netdata' for 1190 seconds.
```

When testing the SSL connection with `openssl s_client -connect parent.node:20000`, I'm getting this:
```
CONNECTED(00000003)
depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
verify return:1
depth=1 C = US, O = Let's Encrypt, CN = R3
verify return:1
depth=0 CN = parent.node
verify return:1
---
Certificate chain
 0 s:CN = parent.node
   i:C = US, O = Let's Encrypt, CN = R3
   a:PKEY: id-ecPublicKey, 384 (bit); sigalg: RSA-SHA256
   v:NotBefore: Dec 16 00:00:00 2022 GMT; NotAfter: Mar 16 00:00:00 2023 GMT
 1 s:C = US, O = Let's Encrypt, CN = R3
   i:C = US, O = Internet Security Research Group, CN = ISRG Root X1
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Sep  4 00:00:00 2020 GMT; NotAfter: Sep 15 16:00:00 2025 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
<SKIPPED>
-----END CERTIFICATE-----
subject=CN = parent.node
issuer=C = US, O = Let's Encrypt, CN = R3
---
No client certificate CA names sent
Peer signing digest: SHA384
Peer signature type: ECDSA
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 2899 bytes and written 402 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 384 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: <SKIPPED>
    Session-ID-ctx:
    Resumption PSK: <SKIPPED>
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - f0 b3 eb 09 4b 2f 6a af-15 fe 44 02 47 e1 03 8a   ....K/j...D.G...
    0010 - 6c a3 7e 10 01 9b 88 ff-7a 6c 5a df 70 78 37 7f   l.~.....zlZ.px7.
    0020 - 5b 2a 33 ae 2f 6b ff 65-87 06 c3 10 b2 c7 14 fc   [*3./k.e........
    0030 - ba 10 1b d4 92 44 95 e5-04 18 86 1d f3 05 d6 fa   .....D..........
    0040 - 79 99 1f e5 6b 30 cc 85-25 d5 32 0e 39 7b e5 b1   y...k0..%.2.9{..
    0050 - f5 12 35 ef fc b7 03 08-76 ef 68 27 26 69 04 bc   ..5.....v.h'&i..
    0060 - e7 2f ba 4d 1a b3 e4 c4-7c f5 f0 a5 5e 4d 4f 1e   ./.M....|...^MO.
    0070 - 1c e8 fa 3f e8 f5 28 40-a2 d4 bf dc 1b bc 36 4b   ...?..(@......6K
    0080 - 0a 6f 1c d0 07 da 52 55-73 03 d5 8d 31 e0 d8 43   .o....RUs...1..C
    0090 - b0 91 a3 8d bf b4 13 e4-0d 4a 49 4c f7 81 8f 74   .........JIL...t
    00a0 - 98 eb 99 5b ee f4 3f 43-2f 4d 64 b0 54 91 3b 71   ...[..?C/Md.T.;q
    00b0 - 48 39 5b e2 6c 7f 16 9c-c4 21 64 70 71 09 1b ad   H9[.l....!dpq...
    00c0 - 0e bc 6f 14 c6 b5 3a 78-19 04 e6 eb 81 37 b2 f5   ..o...:x.....7..

    Start Time: 1671573877
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: <SKIPPED>
    Session-ID-ctx:
    Resumption PSK: <SKIPPED>
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - f0 b3 eb 09 4b 2f 6a af-15 fe 44 02 47 e1 03 8a   ....K/j...D.G...
    0010 - bd b2 c2 40 1a 26 73 a2-c5 35 1e b5 01 f3 94 92   ...@.&s..5......
    0020 - c1 f8 19 00 d9 9d 11 89-25 78 5f ca 08 6e d3 30   ........%x_..n.0
    0030 - 98 a7 59 f3 3b 9a 67 d6-38 8a 25 c3 d1 a0 81 35   ..Y.;.g.8.%....5
    0040 - 78 7c f8 3c d0 71 9c c1-80 ab e3 be 80 65 ca 53   x|.<.q.......e.S
    0050 - cc 29 5c ec fc 2e aa 48-cd 77 ee 30 29 e3 e9 4e   .)\....H.w.0)..N
    0060 - 9e 06 a3 37 c1 6c c3 8b-17 05 f4 20 5d 4d 09 ca   ...7.l..... ]M..
    0070 - 2f 33 8e 8e 0e 88 ab 40-5e ae 61 57 41 96 d4 ca   /3.....@^.aWA...
    0080 - 4a 07 53 58 a0 a3 81 e1-3a 8f fa ac 55 ce 81 43   J.SX....:...U..C
    0090 - d0 2f 95 d5 2d 34 c3 8f-91 bc de a9 c1 d0 c4 86   ./..-4..........
    00a0 - 2c 75 ed aa 1a 38 f8 b1-a1 c0 cb 35 e5 9f fd 16   ,u...8.....5....
    00b0 - c8 5c e0 25 57 78 c0 17-df 25 4f 6f d5 68 f9 ee   .\.%Wx...%Oo.h..
    00c0 - 98 fb 36 3a 14 40 3a 5e-9d 8f 3b 18 68 5a 96 93   ..6:.@:^..;.hZ..

    Start Time: 1671573877
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
40672FA8FB7F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:../ssl/record/rec_layer_s3.c:308:
```

**Comment by MrZammler** [comment]
@Habetdin thank you for this!

@Dim-P can you try the same setup (with the patches), but in netdata.conf set `tls version = 1.2` ?

**Comment by thiagoftsm** [comment]
Hello @MrZammler  and @Dim-P ,

Which distribution and OpenSSL version are you testing? In normal situation we should not downgrade OpenSSL/TLS version.

Best regards!

**Comment by MrZammler** [comment]
> Hello @MrZammler and @Dim-P ,
> 
> Which distribution and OpenSSL version are you testing? In normal situation we should not downgrade OpenSSL/TLS version.
> 
> Best regards!

Hi Thiago. The setup is on FreeBSD current using `openssl-1.1.1q`. Not sure yet why, will do a round of tests/search (other distros as well, with/without patches) and will report.

**Comment by MrZammler** [comment]
@Dim-P Dimitri, is it possible to re-test this on your setup? Switch also to tls 1.3.

**Comment by thiagoftsm** [comment]
Hello @MrZammler ,

Wee have a setback in your PR since I tested it, because right now I have these messages on `CentOS 7.9`:

```sh
 retry in 120 secs, at 2023-01-04 23:05:00
2023-01-04 23:05:00: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain: connecting to '192.168.0.36:19999' (default port: 19999)...
2023-01-04 23:05:00: netdata ERROR : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: remote server rejected this stream, the host we are trying to stream is already streamed to it - will retry in 120 secs, at 2023-01-04 23:07:00
```

and streaming is not working properly with old OpenSSL version.

Best regards!

**Comment by MrZammler** [comment]
Hi Thiago. Ok, will try to test it as well. Is the child on 7.9, parent, or both?

Thanks!

**Comment by MrZammler** [comment]
Hi again Thiago. Just tried a child on Centos 7 with openssl 1.0.2k, streaming to an ubuntu 22, seems to work over ssl (there is a problem with compiling on centos 7, will need an ifdef).

Can we go over your setup when you're here? This particular error might be because of configuration.

**Comment by thiagoftsm** [comment]
Hello @MrZammler ,

The child was `CentOS 7.9` while the parent is `Slackware current`. 
On child I used memory mode `ram`, while on child I used `dbengine`.
I am recompiling branch again to recreate the scenario.

Best regards!

**Comment by MrZammler** [comment]
> Hello @MrZammler ,
> 
> The child was `CentOS 7.9` while the parent is `Slackware current`. On child I used memory mode `ram`, while on child I used `dbengine`. I am recompiling branch again to recreate the scenario.
> 
> Best regards!

Thanks man. Can you check also that they are using different machine guids?

**Comment by thiagoftsm** [comment]
Hello @MrZammler ,

It is possible I mixed branches yesterday, or I had autoconf garbage.  I recreated from scratch the CentOS 7.9 and while I am compiling I received this error:

```sh
[/home/thiago/Netdata/netdata]# make -j1 
web/server/web_client_cache.c: In function 'web_client_reuse_ssl':
web/server/web_client_cache.c:17:13: warning: implicit declaration of function 'SSL_SESSION_is_resumable' [-Wimplicit-function-declaration]
             if (SSL_SESSION_is_resumable(session))
             ^
web/server/web_client_cache.o: In function `web_client_reuse_ssl':
web_client_cache.c:(.text+0xb5): undefined reference to `SSL_SESSION_is_resumable'
collect2: error: ld returned 1 exit status
make[2]: *** [netdata] Error 1
make[1]: *** [all-recursive] Error 1

```

**Comment by MrZammler** [comment]
Indeed, saw that as well, trying to figure out how to ifdef it...

**Comment by thiagoftsm** [comment]
Hello @MrZammler ,

I compiled your branch again and it worked on `CentOS 7.9`, but after I stop `netdata` and  starting it again, I am having the issues:

```sh
2023-01-05 13:53:57: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 13:53:57: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: enabling metrics streaming...
2023-01-05 13:53:57: netdata INFO  : PLUGINSD[apps] : STREAM localhost.localdomain [send]: sending metrics to parent...
2023-01-05 14:08:17: netdata INFO  : MAIN : STREAM localhost.localdomain [send]: signaling sending thread to stop...
2023-01-05 14:08:17: netdata INFO  : MAIN : STREAM localhost.localdomain [send]: waiting for the sending thread to stop...
2023-01-05 14:08:17: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send]: sending thread cleans up...
2023-01-05 14:08:17: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 14:08:17: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send]: sending thread now exits.
2023-01-05 14:08:17: netdata INFO  : STREAM_SENDER[localhost.localdomain] : thread with task id 24472 finished
2023-01-05 14:08:17: netdata INFO  : MAIN : STREAM localhost.localdomain [send]: sending thread has exited.
2023-01-05 14:08:23: netdata INFO  : MAIN : STREAM: added streaming destination No 1: '192.168.0.36:19999' to host 'localhost.localdomain'
2023-01-05 14:08:24: netdata ERROR : GLOBAL_STATS : STREAM localhost.localdomain [send]: not ready - collected metrics are not sent to parent.
2023-01-05 14:08:24: netdata INFO  : STREAM_SENDER[localhost.localdomain] : thread created with task id 25111
2023-01-05 14:08:24: netdata INFO  : STREAM_SENDER[localhost.localdomain] : set name of thread 25111 to STREAM_SENDER[l
2023-01-05 14:08:24: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send]: thread created (task id 25111)
2023-01-05 14:08:24: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain (errno 9, Bad file descriptor)
2023-01-05 14:08:24: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain: attempting to connect to '192.168.0.36:19999' (default port: 19999)...
2023-01-05 14:08:24: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: initializing communication...
2023-01-05 14:08:24: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: waiting response from remote netdata...
2023-01-05 14:08:24: netdata ERROR : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: remote server rejected this stream, the host we are trying to stream is already streamed to it.
2023-01-05 14:08:24: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 14:08:29: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 14:08:29: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain: skipping destination '192.168.0.36:19999' (default port: 19999) due to last error (code: -3, remote server rejected this stream, the host we are trying to stream is already streamed to it), will retry it in 115 seconds
2023-01-05 14:08:29: netdata ERROR : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: could not connect to parent node at this time.
2023-01-05 14:08:34: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 14:08:34: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain: skipping destination '192.168.0.36:19999' (default port: 19999) due to last error (code: -3, remote server rejected this stream, the host we are trying to stream is already streamed to it), will retry it in 110 seconds
2023-01-05 14:08:34: netdata ERROR : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: could not connect to parent node at this time.
2023-01-05 14:08:39: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 14:08:39: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain: skipping destination '192.168.0.36:19999' (default port: 19999) due to last error (code: -3, remote server rejected this stream, the host we are trying to stream is already streamed to it), will retry it in 105 seconds
2023-01-05 14:08:39: netdata ERROR : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: could not connect to parent node at this time.
2023-01-05 14:08:44: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 14:08:44: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain: skipping destination '192.168.0.36:19999' (default port: 19999) due to last error (code: -3, remote server rejected this stream, the host we are trying to stream is already streamed to it), will retry it in 100 seconds
2023-01-05 14:08:44: netdata ERROR : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain [send to 192.168.0.36:19999]: could not connect to parent node at this time.
2023-01-05 14:08:49: netdata ERROR : STREAM_SENDER[localhost.localdomain] : Clearing stream_collected_metrics flag in charts of host localhost.localdomain
2023-01-05 14:08:49: netdata INFO  : STREAM_SENDER[localhost.localdomain] : STREAM localhost.localdomain: skipping destination '192.168.0.36:19999' (default port: 19999) due to last error (code: -3, remote server rejected this stream, the host we are trying to stream is already streamed to it), will retry it in 95 seconds
```

Best regards!

**Review by thiagoftsm — APPROVED** [review]
After last commits all issues related to reconnection and old OpenSSL version were fixed, LGTM!

**Review by Dim-P — APPROVED** [review]
Good job @MrZammler , seems like it also fixes the issue with SSL on local dashboard!

**Pull request merged**